src/runtime/entitlements-manager.js

```diff
@@ -116,6 +116,15 @@ export class EntitlementsManager {
       this.responsePromise_ = this.getEntitlementsFlow_(encryptedDocumentKey);
     }
     return this.responsePromise_.then(response => {
+      if (
+        encryptedDocumentKey != null &&
+        response.decryptedDocumentKey == null
```

Comment:
We also need to check that entitlement is a granting one. It's totally ok to return a `null` decrypted key if the user doesn't have the required access.